lxd/certificates.go

```diff
@@ -439,6 +440,11 @@ func certificatesPost(d *Daemon, r *http.Request) response.Response {
 		return response.NotImplemented(fmt.Errorf("Password authentication is no longer supported, please update your client"))
 	}
 
+	// Validate name.
+	if strings.HasPrefix(req.Name, "--") || strings.HasPrefix(req.Name, "-") {
+		return response.BadRequest(fmt.Errorf("Invalid certificate name"))
+	}
```

Comment:
Do you think `strings.HasPrefix(req.Name, "--")` might be redundant?

Also has there been any discussion around any additional validation we might want to apply. Commonly for lxd entities we disallow forward slashes.